lib/webpack/index.ts

```diff
@@ -91,6 +97,20 @@ module.exports = class WebpackAdapter extends EventEmitter {
   }
 
   startDevServer(server: Server) {
+    const zenConfig = this.zenConfig
+    if (zenConfig?.setDevelopmentHeaders) {
```

Comment:
No changes needed... 
I wish webpackdevserver had a nicer api for a general purpose middleware, instead of us having to hack into the prototypical chain here.

Reply:
Digging further - it looks like this **is effectively** already a full middleware per [the implementation here](https://github.com/webpack/webpack-dev-server/blob/ff2874f5f3a90e5727434cc10f69ac4d54896033/lib/Server.js#L464). 

Newer versions of webpack-dev-server appear to support a full middleware implementation [per here](https://github.com/webpack/webpack-dev-server/blob/d78905bdb0326a7246e2df2fc6a1350acaa4c1b3/lib/Server.js#L2384C34-L2384C50)

I wonder if we could make zen's api match. i.e. `setupMiddlewares` instead of `setDevelopmentHeaders`

----

Feel free to punt.

Reply:
I think I'm going to punt on this on, but I agree the prototype override is not idea. But since we've been doing in desktop for a while we know it works fine